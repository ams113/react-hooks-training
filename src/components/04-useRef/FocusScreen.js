import React, { useRef } from 'react';

export const FocusScreen = () => {

    const inputRef = useRef();

    const handleClick = ( e ) => {
        // document.querySelector('input').select();
        inputRef.current.select();
    };
    
    return (
        <div>
            <h1>FocusScreen</h1>

            <input 
                ref={inputRef}
                type="text" 
                className="form-control"
                placeholder="your name"
            />

            <button 
                className="btn btn-outline-dark mt-5"
                onClick={ handleClick }>
                Focus
            </button>
        </div>
    )
}
