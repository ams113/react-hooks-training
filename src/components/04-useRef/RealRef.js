import React, { useState } from 'react'
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

export const RealRef = () => {


    const [show, setShow] = useState(false);

    return (
        <>
            <div>
                <h1>RealRef</h1>

                { show && <MultipleCustomHooks /> }
            </div>
            <button 
            className="btn btn-dark mt-5"
            onClick={ () => setShow( !show ) }
        >
            Show/Hide
        </button>
    </>
    )
}
