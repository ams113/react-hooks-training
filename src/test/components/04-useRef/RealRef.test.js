import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { RealRef } from '../../../components/04-useRef/RealRef';

describe('Testing Components: RealRef', () => {

    const wrapper = shallow( <RealRef /> );

   
    test('should renders correctly', () => {
         
    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( false );
    
});

test('should show component <MultipleCustomHooks />', () => {
    
    wrapper.find('button').simulate('click');
    expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( true );
  
    });
    
});
