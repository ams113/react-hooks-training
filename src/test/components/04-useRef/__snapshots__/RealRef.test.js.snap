// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Components: RealRef should renders correctly 1`] = `
<Fragment>
  <div>
    <h1>
      RealRef
    </h1>
  </div>
  <button
    className="btn btn-dark mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;
